controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -565,6 +850,24 @@ func (pp *portPublisher) updatePort(targetPort namedPort) {
 	}
 }
 
+func (pp *portPublisher) deleteEndpointSlice(es *discovery.EndpointSlice) {
+	addrSet := pp.endpointSliceToAddresses(es)
+	remove := make(map[ID]Address)
+	for id := range pp.addresses.Addresses {
+		if _, found := addrSet.Addresses[id]; found {
+			delete(pp.addresses.Addresses, id)
+			remove[id] = addrSet.Addresses[id]
+		}
+	}
+
+	for _, listener := range pp.listeners {
+		listener.Remove(AddressSet{remove, addrSet.Labels})
```

Comment:
Do we need to build the `remove` map at all?  I think extraneous removes (removes for addresses not already in the load balancer) will be ignored.  We should just be able to use addrSet here.

Reply:
Yup, on second thought there is no need to build the `remove` map, so not sure why that was in place. Changed it around!